valuable/src/named_values.rs

```diff
@@ -3,17 +3,61 @@ use core::iter::{self, FusedIterator};
 use crate::field::*;
 use crate::*;
 
-/// Access values for a struct's static fields
+/// Set of values from a `Structable` or `Enumerable` with named fields.
 pub struct NamedValues<'a> {
     fields: &'a [NamedField<'a>],
     values: &'a [Value<'a>],
 }
 
 impl<'a> NamedValues<'a> {
+    /// Create a new `NamedValues` instance
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use valuable::{NamedField, NamedValues, Value};
+    ///
+    /// let fields = [
+    ///     NamedField::new("foo"),
+    ///     NamedField::new("bar")
+    /// ];
+    /// let values = [
+    ///     Value::U32(123),
+    ///     Value::U32(456),
+    /// ];
+    ///
+    /// let named_values = NamedValues::new(&fields, &values);
+    ///
+    /// assert_eq!(
+    ///     named_values.get(&fields[0]).unwrap().as_u32(),
+    ///     Some(123));
+    /// ```
     pub fn new(fields: &'a [NamedField<'a>], values: &'a [Value<'a>]) -> NamedValues<'a> {
```

Comment:
Given how `iter()` works, it would probably make sense for this function to require that `fields` and `values` have the same length.

Reply:
:+1: